pkg/apis/objectbucket.io/v1alpha1/objectbucketclaim_types.go

```diff
@@ -1,13 +1,37 @@
 package v1alpha1
 
 import (
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ObjectBucketClaimSpec defines the desired state of ObjectBucketClaim
 type ObjectBucketClaimSpec struct {
+	// StorageClass names the StorageClass object representing the desired provisioner and parameters
 	StorageClass string
+	// BucketName (not recommended) the name of the bucket.  Caution!
+	// In-store bucket names may collide across namespaces.  If you define
+	// the name yourself, try to make it as unique as possible.
+	BucketName string `json:"bucketName,omityempty"`
+	// GenerateBucketName (recommended) a prefix for a bucket name to be
+	// followed by a hyphen and 5 random characters. Protects against
+	// in-store name collisions.
+	GeneratBucketName string `json:"generatBucketName,omitempty"`
+	// SSL whether connection to the bucket requires SSL authentication or not
+	SSL bool `json:"ssl"`
```

Comment:
why is SSL needed in the OBC and OB? I would think this flag is only needed in the OB.

Reply:
Ditto for the` CannedBucketACL`, `Versioned`, and `AdditionalConfig` fields. I would think these field also only part of the OB...

Reply:
> why is `SSL` needed in the OBC and OB? I would think this flag is only needed in the OB.

this comes down to what we want users to be able to request.  Should they be able to ask for an https endpoint via PVC? Or should that be through storage class?

> Ditto for the` CannedBucketACL`, `Versioned`, and `AdditionalConfig` fields. I would think these field also only need to be defined in the OB...

pretty much the same as above